packages/components/src/Menu/MenuItem.tsx

```diff
@@ -24,215 +24,99 @@
-import { useID } from '../utils'
-import { MenuItemContext } from './MenuItemContext'
-import { MenuItemLayout } from './MenuItemLayout'
+import { ListItem, ListItemProps } from '../List'
+import { Icon } from '../Icon'
+import { useForkedRef, useID } from '../utils'
+import { ListItemContext } from '../List/ListItemContext'
+import { listItemDimensions } from '../List/utils'
 import { useNestedMenu, UseNestedMenuProps } from './useNestedMenu'
 
 export interface MenuItemProps
-  extends CompatibleHTMLProps<HTMLLIElement>,
-    Pick<UseNestedMenuProps, 'nestedMenu'> {
-  iconArtwork?: ReactNode
-  compact?: boolean
-  /**
-   * Indicates the MenuItem is checked
-   */
-  current?: boolean
-  /*
-   * optional extra description
-   */
-  description?: ReactNode
-  detail?: ReactNode
-  icon?: IconNames
-  /**
-   * Sets the correct accessible role for the MenuItem:
-   * - Use **'link'** for items that navigation to another page
-   * - Use **'button'** for items that trigger in page interactions, like displaying a dialog
-   * @default 'button'
-   *
-   */
-  itemRole?: 'link' | 'button'
-  tooltip?: string
-  tooltipPlacement?: Placement
-}
-
-const MenuItemInternal: FC<MenuItemProps> = ({
-  children,
-  className,
-  compact: propCompact,
-  current,
-  description,
-  detail,
-  disabled,
-  href,
-  icon,
-  iconArtwork,
-  itemRole,
-  onBlur,
-  onClick,
-  onKeyDown,
-  onKeyUp,
-  onMouseEnter,
-  onMouseLeave,
-  nestedMenu,
-  target,
-  tooltip,
-  tooltipPlacement = 'left',
-  ...props
-}) => {
-  const [isFocusVisible, setFocusVisible] = useState(false)
-
-  const {
-    compact: contextCompact,
-    renderIconPlaceholder,
-    setRenderIconPlaceholder,
-  } = useContext(MenuItemContext)
-  const compact = propCompact === undefined ? contextCompact : propCompact
-
-  const id = useID(props.id)
-
-  const {
-    popover,
-    domProps: { onClick: nestedMenuOnClick, ...nestedMenuProps },
-  } = useNestedMenu({
-    compact,
-    id,
-    nestedMenu,
-    onClick,
-    onKeyDown,
-    onMouseEnter,
-    onMouseLeave,
-  })
-
-  if (detail && nestedMenu) {
-    // eslint-disable-next-line no-console
-    console.warn('The detail prop is not supported when nestedMenu is used.')
-  }
-  detail = nestedMenu ? (
-    <Icon color="text1" name="ArrowRight" size={compact ? 'small' : 'medium'} />
-  ) : (
-    detail
-  )
-
-  const handleOnBlur = (event: React.FocusEvent<HTMLLIElement>) => {
-    setFocusVisible(false)
-    onBlur && onBlur(event)
-  }
-
-  const { closeModal } = useContext(DialogContext)
-
-  const handleOnClick = (event: React.MouseEvent<HTMLLIElement>) => {
-    setFocusVisible(false)
-    // nestedMenuOnClick wraps onClick from props
-    nestedMenuOnClick(event)
-    // Close the Menu unless event has preventDefault
-    if (closeModal && !event.defaultPrevented) {
-      closeModal()
+  extends ListItemProps,
+    Pick<UseNestedMenuProps, 'nestedMenu'> {}
+
+const MenuItemInternal: FC<MenuItemProps> = forwardRef(
```

Comment:
Remove this `MenuItemProps` here and add it below...
```suggestion
const MenuItemInternal = forwardRef(
```

Reply:
Ah I figured I was doing something wrong with Props here since the typing of `ref` on `ListItem` kept erroring out.

Good call! Resolved in af3e87fdf.